src/Main.java

```diff
@@ -12,24 +8,18 @@ public static void main(String[] args) {
         }
         System.out.println("--Enf of task 1-- \n");
         int clientDeviceYear = 2015;
-        if (clientOS == 0) {
-            if (clientDeviceYear >= 2015) {
-                System.out.println("Установите версию приложения для iOS по ссылке ...");
-            } else {
-                System.out.println("Установите облегченную версию приложения для iOS по ссылке ...");
-            }
+        if (clientOS == 0 && clientDeviceYear >= 2015) {
+            System.out.println("Установите версию приложения для iOS по ссылке ...");
+        } else if (clientOS == 0 && clientDeviceYear < 2015 ) {
+            System.out.println("Установите облегченную версию приложения для iOS по ссылке ...");
+        } else if (clientOS == 1 && clientDeviceYear >= 2015) {
+            System.out.println("Установите версию приложения для Android по ссылке ...");
         } else {
-            if (clientDeviceYear >= 2015) {
-                System.out.println("Установите версию приложения для Android по ссылке ...");
-            } else {
-                System.out.println("Установите облегченную версию приложения для Android по ссылке ...");
-            }
+            System.out.println("Установите облегченную версию приложения для Android по ссылке ...");
         }
         System.out.println("--Enf of task 2-- \n");
         int year = 2021;
-        if (year >= 1584 && year % 100 == 0 && year % 400 == 0) {
-            System.out.println(year + " год является високосным.");
-        } else if (year >= 1584 && year % 4 ==0 && year % 100 != 0) {
+        if (year % 400 == 0 || year >= 1584 && year % 100 != 0 && year % 4 == 0) {
```

Comment:
порядок важен, т.к. скобок нет, то приоритет идёт слева направо, причём у И он выше, чем у ИЛИ
```suggestion
        if (year >= 1584 && year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
```